com.unity.render-pipelines.high-definition/Editor/VFXGraph/VFXHDRPBinder.cs

```diff
@@ -14,14 +20,156 @@ class VFXHDRPBinder : VFXSRPBinder
             catch (ArgumentException) // Silently catch the 'Unknown shader' in case of non HDRP shaders
             {}
         }
+
+        public override VFXAbstractRenderedOutput.BlendMode GetBlendModeFromMaterial(VFXMaterialSerializedSettings materialSettings)
+        {
+            var blendMode = VFXAbstractRenderedOutput.BlendMode.Opaque;
+
+            if (!materialSettings.HasProperty(HDMaterialProperties.kSurfaceType) ||
+                !materialSettings.HasProperty(HDMaterialProperties.kBlendMode))
+            {
+                return blendMode;
+            }
+
+            var surfaceType = materialSettings.GetFloat(HDMaterialProperties.kSurfaceType);
+            if (surfaceType == (int)SurfaceType.Transparent)
+            {
+                switch (materialSettings.GetFloat(HDMaterialProperties.kBlendMode))
+                {
+                    case (int)BlendMode.Additive:
+                        blendMode = VFXAbstractRenderedOutput.BlendMode.Additive;
+                        break;
+                    case (int)BlendMode.Alpha:
+                        blendMode = VFXAbstractRenderedOutput.BlendMode.Alpha;
+                        break;
+                    case (int)BlendMode.Premultiply:
+                        blendMode = VFXAbstractRenderedOutput.BlendMode.AlphaPremultiplied;
+                        break;
+                }
+            }
+
+            return blendMode;
+        }
+
+        public override bool TransparentMotionVectorEnabled(Material material)
+        {
+            if (!material.HasProperty(HDMaterialProperties.kSurfaceType) ||
+                !material.HasProperty(HDMaterialProperties.kTransparentWritingMotionVec))
+            {
+                return false;
+            }
+
+            var surfaceType = material.GetFloat(HDMaterialProperties.kSurfaceType);
+
+            if (surfaceType == (int)SurfaceType.Transparent)
+                return material.GetFloat(HDMaterialProperties.kTransparentWritingMotionVec) == 1f;
+
+            return false;
+        }
+
+        public override string GetShaderName(ShaderGraphVfxAsset shaderGraph)
+        {
+            // Recover the HDRP Shader Enum from the VFX Shader Graph.
+            var shaderID = GetShaderEnumFromShaderGraph(shaderGraph);
+
+            if (s_ShaderNames.ContainsKey(shaderID))
+                return s_ShaderNames[shaderID];
+
+            return string.Empty;
+        }
+
+        // List of shader properties that currently are not supported for exposure in VFX shaders.
+        private static readonly Dictionary<Type, string> s_UnsupportedShaderPropertyTypes = new Dictionary<Type, string>()
+        {
+            { typeof(DiffusionProfileShaderProperty), "Diffusion Profile" },
+            { typeof(VirtualTextureShaderProperty),   "Virtual Texture"   },
+            { typeof(GradientShaderProperty),         "Gradient"          }
+        };
+
+        public override bool IsGraphDataValid(GraphData graph)
```

Comment:
This should probably be in the base class because it does not seem HDRP specific

Reply:
But I guess this can be done later when implementing URP